### PR TITLE
Added Product Repository and Use Cases

### PR DESCRIPTION
**Pull request by Shobhith45**

### CODE REVIEW

File: app/src/main/java/com/shobhith/atvproductsapp/home/domain/use_case/GetProductCategories.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.shobhith.atvproductsapp.home.domain.use_case
+
+import com.shobhith.atvproductsapp.home.domain.repository.ProductRepository
+
+class GetProductCategories(private val repository: ProductRepository) {
+    operator fun invoke() =
+        repository.getProductCategories()
+}
```
Comment (vignesh-pai-robosoft) [inline app/src/main/java/com/shobhith/atvproductsapp/home/domain/use_case/GetProductCategories.kt]: Just one suggestion always try to add one blank line at the end of every file. Now it's not an issue. But lint will throw an warning.

File: app/src/main/java/com/shobhith/atvproductsapp/home/domain/use_case/GetProductByCategoryName.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.shobhith.atvproductsapp.home.domain.use_case
```
Comment (vignesh-pai-robosoft) [inline app/src/main/java/com/shobhith/atvproductsapp/home/domain/use_case/GetProductByCategoryName.kt]: let the package name be simple without any underscore. like usecase. use_case not required

**Review by vignesh-pai-robosoft — COMMENTED** [review]
Rest everything LGTM.

File: app/src/main/java/com/shobhith/atvproductsapp/home/domain/use_case/GetProductCategories.kt (hunk 1 above)
Comment (Shobhith45) [reply app/src/main/java/com/shobhith/atvproductsapp/home/domain/use_case/GetProductCategories.kt]: Fixed.

File: app/src/main/java/com/shobhith/atvproductsapp/home/domain/use_case/GetProductByCategoryName.kt (hunk 2 above)
Comment (Shobhith45) [reply app/src/main/java/com/shobhith/atvproductsapp/home/domain/use_case/GetProductByCategoryName.kt]: Fixed.

**Review by vignesh-pai-robosoft — APPROVED** [review]

**Pull request merged**